Build fails - not linking dl on Arch Linux

Building with **clang 3.5** and **cmake 3.4**.

Errors from building look like flag -ldl is not passed to compiler/linker so it does not link with dl library, which is needed on Linux to use dynamic linking(`dlopen` etc.).

I've grep'ed source code to see if you link to dl, and it wasn't there. ```grep -ri target_link_libraries | grep -w dl``` found nothing.

This is Error:

```
../unixcoreruncommon/libunixcoreruncommon.a(coreruncommon.cpp.o): In function `ExecuteManagedAssembly(char const*, char const*, char const*, int, char const**)':
/build/coreclr-git/src/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:270: undefined reference to `dlopen'
/build/coreclr-git/src/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:273: undefined reference to `dlsym'
/build/coreclr-git/src/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:274: undefined reference to `dlsym'
/build/coreclr-git/src/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:275: undefined reference to `dlsym'
/build/coreclr-git/src/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:369: undefined reference to `dlclose'
/build/coreclr-git/src/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:376: undefined reference to `dlerror'
../unixcoreruncommon/libunixcoreruncommon.a(coreruncommon.cpp.o): In function `ExecuteManagedAssembly(char const*, char const*, char const*, int, char const**)':
/build/coreclr-git/src/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:270: undefined reference to `dlopen'
/build/coreclr-git/src/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:273: undefined reference to `dlsym'
/build/coreclr-git/src/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:274: undefined reference to `dlsym'
/build/coreclr-git/src/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:275: undefined reference to `dlsym'
/build/coreclr-git/src/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:369: undefined reference to `dlclose'
/build/coreclr-git/src/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:376: undefined reference to `dlerror'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
src/coreclr/hosts/unixcoreconsole/CMakeFiles/coreconsole.dir/build.make:95: recipe for target 'src/coreclr/hosts/unixcoreconsole/coreconsole' failed
make[2]: *** [src/coreclr/hosts/unixcoreconsole/coreconsole] Error 1
CMakeFiles/Makefile2:62041: recipe for target 'src/coreclr/hosts/unixcoreconsole/CMakeFiles/coreconsole.dir/all' failed
make[1]: *** [src/coreclr/hosts/unixcoreconsole/CMakeFiles/coreconsole.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
src/coreclr/hosts/unixcorerun/CMakeFiles/corerun.dir/build.make:95: recipe for target 'src/coreclr/hosts/unixcorerun/corerun' failed
make[2]: *** [src/coreclr/hosts/unixcorerun/corerun] Error 1
CMakeFiles/Makefile2:61985: recipe for target 'src/coreclr/hosts/unixcorerun/CMakeFiles/corerun.dir/all' failed
make[1]: *** [src/coreclr/hosts/unixcorerun/CMakeFiles/corerun.dir/all] Error 2
```

--- a/src/coreclr/hosts/unixcoreruncommon/CMakeLists.txt
+++ b/src/coreclr/hosts/unixcoreruncommon/CMakeLists.txt
@@ -5,3 +5,4 @@ add_library(unixcoreruncommon
     coreruncommon.cpp
 )
 
+target_link_libraries(unixcoreruncommon dl)

--- a/src/coreclr/hosts/unixcoreruncommon/CMakeLists.txt
+++ b/src/coreclr/hosts/unixcoreruncommon/CMakeLists.txt
@@ -5,4 +5,6 @@ add_library(unixcoreruncommon
     coreruncommon.cpp
 )
 
-target_link_libraries(unixcoreruncommon dl)
+if(CLR_CMAKE_PLATFORM_LINUX)
+  target_link_libraries(unixcoreruncommon dl)
+endif(CLR_CMAKE_PLATFORM_LINUX)

